gnuplot/src/color.rs

```diff
@@ -0,0 +1,413 @@
+pub use self::ColorType::*;
+use crate::util::OneWayOwned;
+use std::fmt::Display;
+
+pub trait IntoColor<T>: Into<ColorType<T>> + Clone {}
+impl<TC, T: Into<ColorType<TC>> + Clone> IntoColor<TC> for T {}
+
+pub type ColorIndex = u8;
+pub type ColorComponent = u8;
+pub type ColorInt = u32;
+pub type RGBInts = (ColorComponent, ColorComponent, ColorComponent);
+pub type ARGBInts = (
+	ColorComponent,
+	ColorComponent,
+	ColorComponent,
+	ColorComponent,
+);
+
+/// Option type (for plots, borders, and text) that allows the various different gnuplot
+/// color formats. The gnuplot [colorspec reference](http://gnuplot.info/docs_6.0/loc3640.html)
+/// also explains these.
+///
+/// There are many equivalent ways of specifying colors, and this allows the user to chose the most convenient.
+/// For example, all the following will produce the same blue color:
+/// `RGBColor("blue".into())`, `RGBColor("0x0000ff".into())`, `RGBColor("#0000ff".into())`, `RGBColor("0x000000ff".into())`,
+/// `RGBColor("#000000ff".into())`, `RGBIntegerColor(0, 0, 255)`, `ARGBColor(0, 0, 0, 255)`,
+///
+/// See example usages of these colors in `color.rs` and `variable_color.rs` in the
+/// [Examples folder](https://github.com/SiegeLord/RustGnuplot/tree/master/gnuplot/examples) on Github
+#[derive(Debug, Clone, PartialEq, PartialOrd)]
+pub enum ColorType<T = String>
+{
+	/// string (`&str` or `String`, but usually created with `&str`) in one of the following gnuplot-supported formats
+	/// - colorname   --- e.g. "blue" [See the gnuplot
+	///     [list of colornames](http://gnuplot.info/docs_6.0/loc11229.html)]
+	/// - 0xRRGGBB    --- string containing hexadecimal constant
+	/// - 0xAARRGGBB  --- string containing hexadecimal constant
+	/// - #RRGGBB     --- string containing hexadecimal in x11 format
+	/// - #AARRGGBB   --- string containing hexadecimal in x11 format
+	///
+	/// "#AARRGGBB" represents an RGB color with an alpha channel (transparency) value in the high bits.
+	/// An alpha value of 0 represents a fully opaque color; i.e., "#00RRGGBB" is the same as "#RRGGBB".
+	/// An alpha value of 255 (FF) represents full transparency.
+	RGBString(T),
+	/// tuple of u8 representing red, green and blue values as 0-255
+	RGBInteger(ColorComponent, ColorComponent, ColorComponent),
+	/// tuple of u8 representing alpha, red, green and blue values as 0-255.
+	/// As with `RGBColor`, an alpha value of 0 represents a fully opaque color;
+	/// an alpha value of 255 (FF) represents full transparency.
```

Comment:
It's very upsetting that gnuplot defines alpha backwards here (everywhere else, alpha = 0 -> fully transparent, alpha = 255 -> fully opaque). I wonder if the Rust API should invert it. I'll need to think about it.

Reply:
Yes, this is a tricky one. Both options could lead to "unexpected behaviour". The fix of inverting it could be done cleanly in the `data()` function, if that's the route you decide to go down.